auslib/test/admin/views/test_rules.py

```diff
@@ -141,9 +141,22 @@ def testNewRuleWithoutPermission(self):
     def testMissingFields(self):
         # But we still need to pass product, because permission checking
         # is done before what we're testing
-        ret = self._post('/rules', data=dict({'product': 'a'}))
+        post_data = {'product': 'a'}
+        ret = self._post('/rules', data=post_data)
+        self.assertEquals(ret.status_code, 400, "Status Code: %d, Data: %s" % (ret.status_code, ret.data))
+        # Connexion and manual request.json validation methods will preempt and throw 400 error as
+        # soon as the first field invalidates. Only one field may be present in the response.data depending
+        # upon the order of fields sorted in swagger yaml file and in views.
+        self.assertTrue('title' in ret.data, msg=ret.data)
```

Comment:
Thanks for the insight @tarekziade.

@aksareen - I wonder if it's worth asking the jsonschema folks if there's a way to get more specific information about errors? Or maybe looking through their issue tracker? https://github.com/Julian/jsonschema/issues/119 talks about a "schema_path" and "path" in the Exception object, maybe those might help?